src/fluree/db/json_ld/shacl.cljc

```diff
@@ -561,6 +573,21 @@
                                                                property-shape)
                                              p-shapes*      (update p-shapes path util/conjv property-shape*)
                                              ;; elevate following conditions to top-level custom keys to optimize validations when processing txs
+                                             class-iris      (when-let [class-sids (:class property-shape)]
+                                                               (let [id-path (fn [sid] [sid const/iri-id])]
+                                                                 (loop [[csid & csids] class-sids
+                                                                        ciris []]
+                                                                   (let [ciri (->> csid
+                                                                                   id-path
+                                                                                   (query-range/index-range db :spot =)
+                                                                                   <?
+                                                                                   first
+                                                                                   flake/o)
```

Comment:
Unwinding this thread-last and dropping the `id-path` fn makes it a lot easier for me to understand what the actual query is:
```clojure
(->> (<? (query-range/index-range db :spot = [csid const/iri-id]))
        first
        flake/o)
```

Reply:
Yeah in retrospect that `id-path` fn is silly. Removing it.